src/client/browser/service-worker.js

```diff
@@ -0,0 +1,47 @@
+const MAX_RETRY   = 10;
```

Comment:
Write the new files with TypeScript.

Reply:
At this moment, only server-side scripts are written in TS. It is not an easy task to enable TS for client scripts in the context of the current issue

Reply:
Ok. Let's make it a separate PR.

Reply:
>At this moment, only server-side scripts are written in TS. 
this my statement is wrong. However, it is still not easy in the context of this PR. So, yes, I would prefer to make separate PR for this